### PR TITLE
more forgiving regex for style tags

### PR DESCRIPTION
**Pull request by alanmbarr**
support for variants of style tag such as <style type="text/css"></style>

### CODE REVIEW

**Comment by ahstro** [comment]
Could you wrap your example in backticks (\` or ```)? :)

**Comment by alanmbarr** [comment]
`<style type="text/css"></style>` related to issue 1705

File: seed/challenges/html5-and-css.json [hunk 1]
```diff
@@ -384,7 +384,7 @@
         "assert($('h2').css('color') === 'rgb(0, 0, 255)', 'Your <code>h2</code> element should be blue.')",
         "assert(!$('h2').attr('style'), 'Remove the style attribute from your <code>h2</code> element.')",
         "assert(($('style').length > 1), 'Create a <code>style</code> element.')",
-        "assert(editor.match(/<\\/style>/g) && editor.match(/<\\/style>/g).length === editor.match(/<style>/g).length, 'Make sure each of your <code>style</code> elements has a closing tag.')"
+        "assert(editor.match(/<style.*>.*<\\/style>/g) && editor.match(/<style.*>.*<\\/style>/g).length === editor.match(/<style.*>.*<\\/style>/g).length, 'Make sure each of your <code>style</code> elements has a closing tag.')"
       ],
```
Comment (BerkeleyTrue) [inline seed/challenges/html5-and-css.json]: This won't work.

This check will always be true:
```js
... && editor.match(/<style.*>.*<\/style>/g).length === editor.match(/<style.*>.*<\/style>/g).length
```
Its checking to see if the number of style elements is equal to the number of style elements, or `1=1` in this case.

The import thing is to make `editor.match(/<style>/g)` allow for an optional type attribute equal to 'text/css'.

**Comment by alanmbarr** [comment]
Thanks for looking into this. I'll close this for now until I can come up with a better solution.

**Pull request closed without merging**